### DISCUSSION TITLE
Game Freezing

### DISCUSSION THREAD

**Discussion by allama699** [LizardByte/community#328] [Sunshine]
Hi all. This has probably been done to death, but none of the threads have solved my issues. I have recently moved from Steam client on TV over to Sunshine/Moonlight on an Nvidia Shield. This works well for the majority of games, but for a few i get intermittent freezing. I have followed the suggestions regarding HAG etc, but seems to have no affect. I have also read that the freezes for some don't affect the sound. Mine however affects both. I am running a very beefy rig. GPU is a 4090, CPU is an AMD 7950X3d. Gen 4 M.2 SSD and 64 GIG of DDR5 5600 RAM. The network is a dedicated 1Gbit wired setup. Oh and I'm running Windows 11,  Latest drivers and BIOS. 

FYI.  If i run the affected games via the Steam app on my TV, I get no freezes or stutters. But accessing  Steam via Moonlight on Nvidia Shield does freeze. Nvidia Shield is brand new, and the latest version. I factory reset and re-installed from scratch as well.

Is this a common issue? Is it related to Windows 11? Any nugget of hope would be great.

Thanks in advance 

Andy

**Comment by caron2k** [comment]
try this: Configuration -> NVIDIA NVENC Encoder -> Use realtime priority in hardware accelerated gpu scheduling  to Disabled

**Reply by caron2k** [reply]
I have encountered this problem while playing the game(Alan Wake 2,  just this one), and after setting it up this way, it was solved.  i7+3070+win11

**Reply by brandonSc** [reply]
This results in a really choppy framerate for me.

**Comment by allama699** [comment]
I have disabled GPU scheduling, and disabled realtime priority. Makes no difference to my issue. The sound sometimes drops, and sometime the sound and games freezes for about 1-2 seconds. this happens every 10 mins or so. I have lowered graphical settings but seems to make no difference. Games affected. Alan Wake 2, Diablo 4(Not as much), Star Citizen is the worst, system Shock remake, and Lords of the fallen remake.

**Comment by maj0r-zero** [comment]
Unfortunately this issue affects me too in Alan Wake 2. Setting "Use realtime priority in hardware accelerated gpu scheduling" to Disabled fixes the freezing, but it becomes a stutter fest. Game crashes mostly in menus.

**Reply by maj0r-zero** [reply]
Disabling GPU Scheduling in Windows also solves this issue, of course. But Frame Generation becomes unavailable.

**Comment by brandonSc** [comment]
I have been getting very similar issues for a while now, on most (all?) games I play, however, it seems to have become more troublesome on the latest v0.21.0 release. 

On the previous version, I would simply just reconnect the client if the stream froze, and then it would continue working again. Now it seems that I need to restart Sunshine on my PC, otherwise the clients are unable to reconnect.

**Comment by aacebedo** [comment]
Same on my side on 4080S. It seems that Frame generation is pretty taxing on GPU and result in either image freeze (when real-time priority is used) or stutter fest (sub 30fps) on Alan wake 2 and also Spiderman.

**Comment by 1stwice** [comment]
Same issue here on Version 2025.924.154138. Screen freezes but audio comes through. Have to close and open the stream each time. Disabled HAGS in windows and unchecked realtime priority in sunshine. Anyone found a solution yet?

**Comment by GhostShellLabs** [comment]
Hey all — adding some deeper-level info that might help narrow this down.

What you’re describing (video + audio freezing for 1–2 seconds at regular-ish intervals) is almost always tied to scheduling stalls in Windows 11’s GPU pipeline, especially when NVENC + HAGS + FG (Frame Generation) overlap on the same scheduling boundary.

A few things worth trying that typically don’t show up in the usual “disable HAGS” threads:

1. Check for NVENC Overruns (Windows 11 bug + 40-series cards)

On 4090/4080/4080S, Windows 11 can briefly stall NVENC when:

A game uses Frame Generation (FG)

NVENC attempts real-time submission

The OS switches scheduling contexts

This causes exactly the 1–2 second freeze you’re describing, while Steam Link works fine because Steam uses a different encoder path.

Fix to test:

Sunshine → NVIDIA NVENC Encoder

Disable Use low-latencyquality mode

Set VBV Buffer to a fixed 4–6 MB

Turn off Look-ahead

These reduce NVENC stall risk significantly.

2. Force Windows to stop parking E-cores (7950X3D quirk)

Windows sometimes decides to migrate game threads or NVENC’s CPU-side prep threads onto efficiency cores during long sessions. When this happens mid-stream, you get a hard freeze.

Try:

BIOS → Disable Core Parking / CPPC Preferred Cores / Global C-states

Or Windows → High Performance (not Balanced)

Optional: Use Process Lasso to keep sunshine.exe + game on preferred cores

A surprising number of Moonlight freeze issues come from Ryzen core-migration delays.

3. Audio freezing too means this is NOT network — it’s a system stall

When video freezes but audio continues → network pacing issue
When both freeze → GPU or OS-level stall.

That matches:

HAGS bugs

FG conflicts

NVENC wait stalls

Queue overruns in the Windows Video/Audio shared timestamp clock

4. Try limiting Sunshine to 120 FPS or 90 FPS

If your Shield is doing frame pacing + your PC is generating FG frames on top of native FPS, the encoded framerate can spike.

Try setting:

Sunshine → Max FPS: 120

NVENC → Quality preset P4 (better scheduling behavior)

This alone fixes the issue for many AW2, Diablo IV, and Spider-Man players.

5. Check for MPO (Multiplane Overlay) conflicts

MPO + Windows 11 + new NVIDIA drivers are notorious for causing periodic stalls.

Quick test:

Registry → Disable MPO

[HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows\Dwm]
"OverlayTestMode"=dword:00000005


Reboot → test.

If freezing disappears → driver/MPO conflict confirmed.

6. Steam Link working fine is actually a BIG clue

Steam bypasses:

Sunshine’s NVENC settings

Real-time priority settings

Some scheduling decisions

The FG → NVENC pipeline interaction

So the fact that Steam Link works heavily points toward:

NVENC

HAGS

Scheduling

FG

or MPO interactions

Not your network.

If you want the short version:

Moonlight + Windows 11 + HAGS + Frame Generation + NVENC is a fragile combination right now. The freeze you’re seeing is a known interaction at the OS scheduling layer.

Try in this order:

Disable Look-ahead

Set fixed VBV 4–6MB

Limit Sunshine to 120 FPS

Disable MPO

Force Sunshine/game threads onto performance cores

One of these usually resolves it.

Let us know if any of the above changes the behavior — these kinds of reports help narrow down whether it’s NVENC stalls or scheduling regression in recent drivers.
